### PR TITLE
Feature/cloud 911 s3

### PR DESCRIPTION
**Pull request by rchengvariant**
# Description

Created s3 module with ignore changes to replication rules.

Fixes [#911](https://drivevariant.atlassian.net/browse/CLOUD-911https://usxtech.atlassian.net/browse/CLOUD-8)

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [ ] Sanity Testing
```yaml
Pre reqs:
  - Create terraform.tfvars.json (provided in readme.md)
  - Create provider.tf (provided in readme.md)
  - In version.tf, change (backend s3) -> (backend local)
  - Clone/Pull branch
  - Export DPL Account variables
 
Test Case 1 [Create S3 bucket locally]:
- Steps
  - Terraform validate
  - Terraform init
  - Terraform plan/apply
- Result
 - See Output Bucket name and Bucket arn in terminal
 - See Bucket Created in dpl

Test Case 2 [Destroy S3 Bucket locally]:
- Steps
  - terraform destroy
- Results
  - See Bucket deleted

Test Case 3 [Create S3 Bucket with backend s3]:
- Steps
  - In version.tf, change (backend local) -> (backend s3)
  - remove statefiles/lock/.terraform
  - terraform init \
      -backend-config="bucket=test-bucket-156gpjfbx4m63stm" \
      -backend-config="key=bucket/test" \
      -backend-config="region=us-west-2" \
      -backend-config="dynamodb_table=lazy_tf_state" \
      -backend-config="encrypt=true"
  - terraform plan/apply
- Results
 - See Output Bucket name and Bucket arn in terminal
 - See Bucket Created in dpl
 - See State Key for CREATED BUCKET in S3 Bucket test-bucket-156gpjfbx4m63stm
 - See State Lock for CREATED BUCKET in lazy_tf_state Dynamo DB


Test Case 4 [Delete S3 Bucket with backend s3]:
- Steps
  - terraform destroy
- Results
  - See Bucket deleted
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

File: s3/main.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  source = "github.com/variant-inc/lazy-terraform//submodules/tags?ref=v1"
+
+  name = local.name
+
+  user_tags    = var.user_tags
+  octopus_tags = var.octopus_tags
+}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket = local.name
```
Comment (vibindaniel) [inline s3/main.tf:23]: You can use bucket_prefix instead and provide the `var.bucket_prefix`

**Comment by LeeHughsVariant** [comment]
bucket has default encryption and tags:
image

**Review by LeeHughsVariant — CHANGES_REQUESTED** [review]
Only requested changes is to update the readme to include specific instructions/example of using this module as a module. IE
```
module "aws_s3" {
  source = "github.com/variant-inc/terraform-aws-s3//s3?ref=v1"
  ...
```
Also request to have the default tags in the readme changed. Purpose should not be "dms testing". 
Other than that, looks good to me.

**Comment by naveen-variant** [comment]
Block public access is missing with module . Adding example configuration added lazy api s3.

       Bucket=bucket_name,
        PublicAccessBlockConfiguration={
            'BlockPublicAcls': True,
            'IgnorePublicAcls': True,
            'BlockPublicPolicy': True,
            'RestrictPublicBuckets': True
        },

**Comment by rchengvariant** [comment]
@LeeHughsVariant Updated ReadMe with module example. Please check if it's accurate. Also updated purpose in user tags.

@naveen-variant Correct me if I'm wrong, but setting `acl = "private"` already blocks all public access. I believe further configuration is not necessary.

**Pull request merged**